sources/osgShader/Compiler.js

```diff
@@ -945,6 +955,34 @@ define( [
             for ( var j = 0, jl = nodes.length; j < jl; j++ ) {
                 this.traverse( func, nodes[ j ] );
             }
+
+
+            // Attribute 0 Must Be vertex
+            // perf warning in console otherwiser in opengl Desktop
+            if ( func._text.length ) {
+                // sort in alphabetical order
+                // attr, unif, sample, varying
+                func._text.sort();
+                // now sort Attributes
+                // making sure Vertex is always coming first
+                var toShift = [];
+                for ( j = 0; j < func._text.length && func._text[ 0 ].indexOf( 'Vertex' ) === -1; j++ ) {
```

Comment:
it's hard to read 
j < func._text.length && func._text[ 0 ].indexOf( 'Vertex' )
could simplify the logic or split into multiline